cleverpush/src/main/java/com/cleverpush/NotificationOpenedActivity.java

```diff
@@ -1,27 +1,38 @@
 package com.cleverpush;
 
-import static com.cleverpush.Constants.LOG_TAG;
+import static com.cleverpush.NotificationOpenedProcessor.processIntent;
 
 import android.app.Activity;
 import android.content.Intent;
 import android.os.Bundle;
-import com.cleverpush.util.Logger;
 
 public class NotificationOpenedActivity extends Activity {
 
+    public static NotificationOpenedActivity notificationOpenActivity = null;
```

Comment:
Remove

Reply:
Right now we are using this to pass actual activity as variable

> NotificationOpenedActivity.notificationOpenActivity = activity;

So when the customer finishes the activity from the callback listener it finishes correct activity - (We have also checked with @cpKevinrodriguesnobre 

```
CleverPush.getInstance(this).init(
                getString(R.string.channel_id),
                (NotificationReceivedListener) result -> System.out.println("Received CleverPush Notification: " + result.getNotification().getTitle()),
                (NotificationOpenedListener) (result, appActivity) -> {
                    System.out.println("Opened CleverPush Notification: " + result.getNotification().getUrl());
                    appActivity.finish();
                },
                subscriptionId -> System.out.println("CleverPush Subscription ID: " + subscriptionId));
```